Sources/SpeziViews/Views/MarkdownDocument/MarkdownDocumentView.swift

```diff
@@ -0,0 +1,84 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2025 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import MarkdownUI
+import SpeziFoundation
+import SwiftUI
+
+
+public struct MarkdownDocumentView<CustomElementView: View>: View {
+    public struct DividerRule {
+        @inlinable public static var never: Self {
+            .init { _, _ in false }
+        }
+        
+        @inlinable public static var always: Self {
+            .init { _, _ in true }
+        }
+        
+        @inlinable
+        public static func custom( // swiftlint:disable:this type_contents_order
+            _ shouldInsertDivider: @escaping @MainActor (_ fstBlockIdx: Int, _ sndBlockIdx: Int) -> Bool
+        ) -> Self {
+            .init(shouldInsertDivider: shouldInsertDivider)
+        }
+        
+        fileprivate let shouldInsertDivider: @MainActor (_ fstBlockIdx: Int, _ sndBlockIdx: Int) -> Bool
+        
+        @usableFromInline
+        init(shouldInsertDivider: @escaping @MainActor (_ fstBlockIdx: Int, _ sndBlockIdx: Int) -> Bool) {
+            self.shouldInsertDivider = shouldInsertDivider
+        }
+    }
+    
+    
+    private let markdownDocument: MarkdownDocument
+    private let dividerRule: DividerRule
+    private let customElementViewProvider: @MainActor (_ blockIdx: Int, _ element: MarkdownDocument.CustomElement) -> CustomElementView
+    
+    public var body: some View {
+        let blocks = markdownDocument.blocks
+        VStack(spacing: 12) {
+            ForEach(Array(blocks.indices), id: \.self) { blockIdx in
+                let block = blocks[blockIdx]
+                view(for: block, at: blockIdx)
+                    .id(block.id)
+                let isLast = blockIdx >= blocks.endIndex - 1
+                if !isLast && dividerRule.shouldInsertDivider(blockIdx, blockIdx + 1) {
+                    Divider()
+                }
+            }
+        }
+    }
+    
+    
+    public init(
+        markdownDocument: MarkdownDocument,
+        dividerRule: DividerRule,
+        @ViewBuilder customElementViewProvider: @escaping @MainActor (_ blockIdx: Int, _ element: MarkdownDocument.CustomElement) -> CustomElementView
```

Comment:
Probably wondering if we should provide default values for both of these parameters? E.g., dividerRule can be `.never` by default and `customElementViewProvider` would basically be an empty view mapper that would put an warning/error in the console when there is an actually custom element that should be parsed?

Reply:
the way it currently works is that we have an initializer where you don't need to specify the `customElementViewProvider` and it simply defaults everything that isn't markdown to an `EmptyView`; i agree that we probably should log smth to the console as well

Reply:
pushed the commit. we now only have a single init (apart from the legacy ones for matching the previous `MarkdownView`'s API), which requires the `markdownDocument` be present and defaults both the `dividerRule` and the `customElementViewProvider`

Reply:
Thank you!